core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,180 @@
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (job.ServiceCtx, error) {
 	// TODO (CAPPL-22, CAPPL-24):
 	//   - decode config
 	//   - create an implementation of the capability API and add it to the Registry
 	//   - create a handler and register it with Gateway Connector
 	//   - manage trigger subscriptions
 	//   - process incoming trigger events and related metadata
-	return nil, nil
+
+	handler := &workflowConnectorHandler{
+		connector: connector,
+		signerKey: signerKey,
+		lggr:      lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	// is this the right way to register with gateway connector?  Cron trigger doesn't do this.
+	err := connector.AddHandler([]string{"add_workflow"}, handler)
+
+	return handler, err
+}
+
+// from Web API Trigger Doc
+// trigger_id          - ID of the trigger corresponding to the capability ID
+// trigger_event_id    - uniquely identifies generated event (scoped to trigger_id and sender)xx
+// timestamp           - timestamp of the event, needs to be within certain freshness to be processed
+// sub_events {        - a list of per-topic-set components of this trigger event (can be of size 1)
+//   topics            - [OPTIONAL] list of topics (strings) to be started by this event (affects all topics if empty)
+//   workflow_owners   - [OPTIONAL] list of workflow owners allowed to receive this event (affects all workflows if empty)
+//   params            - key-value pairs that will be used as trigger output in the workflow Engine (translated to values.Map)
+
+//  The sample script does
+// 	workflowSpec := flag.String("workflow_spec", "[my spec abcd]", "Workflow Spec")
+// 	payloadJSON := []byte("{\"spec\": \"" + *workflowSpec + "\"}")
+//
+// how do these reconcile?
+// How do we get the TriggerID to look up in the map of TriggerIDs to connection?
+
+func (h *workflowConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+	body := &msg.Body
+	fromAddr := ethCommon.HexToAddress(body.Sender)
+	// TODO: apply allowlist and rate-limiting
+	h.lggr.Debugw("handling gateway request", "id", gatewayID, "method", body.Method, "address", fromAddr)
+	// ERR: payload is error, how to unmarshall?
+	payload := body.Payload.UnmarshalJSON(body.Payload)
+	triggerID := payload.trigger_id
+
+	switch body.Method {
+	case workflow.MethodAddWorkflow:
+		// TODO: add a new workflow spec and return success/failure
+		// we need access to Job ORM or whatever CLO uses to fully launch a new spec
+		h.lggr.Debugw("added workflow spec", "payload", string(body.Payload))
+
+		// Question: should this call the registerTrigger and then handleNodeResponse call unregister?
+		tr := capabilities.TriggerResponse{
+			Event: capabilities.TriggerEvent{
+				TriggerType: "__builtin_web-api-trigger",
+				ID:          triggerID,
+				Outputs:     payload,
```

Comment:
Here you'll need to JSON-parse the received payload, convert it to a values.Map, then match topics to registered workflows and send TriggerEvent messages accordingly. You can do that in a separate PR if you want.